File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/JoinRequest.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.dto.domain.InviteCode;

public class JoinRequest extends BaseRequest {

  private final InviteCode inviteCode;

  @JsonCreator
  public JoinRequest(@JsonProperty("inviteCode") InviteCode inviteCode) {
    super(MessageType.JOIN_REQUEST);
    this.inviteCode = inviteCode;
  }

  public InviteCode getInviteCode() {
    return inviteCode;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/inbound/JoinRequest.kt
package com.yscorp.withpush.messagesystem.dto.websocket.inbound

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.dto.domain.InviteCode


class JoinRequest @JsonCreator constructor(@JsonProperty("inviteCode") val inviteCode: InviteCode) :
    BaseRequest(MessageType.JOIN_REQUEST) {
}
